r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/codec/BooleanCodec.java

```diff
@@ -38,7 +40,33 @@ private BooleanCodec() {
     @Override
     public Boolean decode(ByteBuf value, MySqlReadableMetadata metadata, Class<?> target, boolean binary,
         CodecContext context) {
-        return binary || metadata.getType() == MySqlType.BIT ? value.readBoolean() : value.readByte() != '0';
+        MySqlType dataType = metadata.getType();
+
+        if (dataType == MySqlType.VARCHAR) {
+            if (!value.isReadable()) {
+                return createFromLong(0);
+            }
+
+            String s = value.toString(metadata.getCharCollation(context).getCharset());
+
+            if (s.equalsIgnoreCase("Y") || s.equalsIgnoreCase("yes") ||
+            s.equalsIgnoreCase("T") || s.equalsIgnoreCase("true")) {
+                return createFromLong(1);
+            } else if (s.equalsIgnoreCase("N") || s.equalsIgnoreCase("no") ||
+            s.equalsIgnoreCase("F") || s.equalsIgnoreCase("false")) {
+                return createFromLong(0);
+            } else if (s.matches("-?\\d*\\.\\d*") || s.matches("-?\\d*\\.\\d+[eE]-?\\d+")) {
+                return createFromDouble(Double.parseDouble(s));
+            } else if (s.matches("-?\\d+")) {
+                if (!CodecUtils.isGreaterThanLongMax(s)) {
+                    return createFromLong(CodecUtils.parseLong(value));
+                }
+                return createFromBigInteger(new BigInteger(s));
+            }
+            throw new IllegalArgumentException("Unable to interpret string: " + s);
```

Comment:
For this type of error, `R2dbcNonTransientResourceException` is appropriate. Could you also include the type in the message? like "The value '%s' of type '%s' cannot be decoded into a Boolean." with sqlState: '22018'